### DISCUSSION TITLE
OGP is not displayed when dynamically adding meta tags using react-helmet-async

### DISCUSSION THREAD

**Discussion by SuzukiMika-dev** [vercel/next.js#72646] [Help]
### Summary

Thank you for your confirmation. I am Japanese, so please excuse any awkward translations.   
The issue I'm facing is that the OGP is not being displayed.  

### Environment
Next 15.0.2  
React 18.3.1  
node 20.9.0  
react-helmet-async 2.0.5   

### Problem
What I want to achieve is to have the OGP displayed when the site URL is pasted into Slack.  
Currently, I'm using react-helmet-async to dynamically add meta tags, and I've confirmed that the OGP is displayed on this site: https://rakko.tools/tools/9/.   
However, it's not displayed on the following sites:  
- https://web-toolbox.dev/tools/ogp-checker  
- https://cards-dev.x.com/validator  

I would greatly appreciate if someone with knowledge could advise me on how to make the OGP display on all sites.

```app/page.tsx
import React from "react";
import { Helmet } from "react-helmet-async";

export const getServerSideProps = async () => {
  const [res] = await Promise.all([
    fetch(
      `https://gist.githubusercontent.com/SuzukiMika-dev/55c0279af147f2f4ca2da5c2e7c8d084/raw/meta-movie-info.json`
    ),
  ]);
  if (!res.ok) {
    const errorText = await res.text();
    console.error("Error fetching data:", errorText);
    return;
  }
  const data = await res.json();

  return {
    props: {
      metaData: data,
    },
  };
};

const Top: React.FC<{ metaData: any }> = ({ metaData }) => {
  return (
    <>
      <Helmet>
        <title>{metaData.title}</title>
        <meta name="description" content={metaData.description} />
        <meta property="og:site_name" content={metaData["og:site-name"]} />
        <meta property="og:image" content={metaData["og:image"]} />
        <meta property="og:title" content={metaData["og:title"]} />
        <meta property="og:description" content={metaData["og:description"]} />
        <meta property="og:url" content={metaData["og:url"]} />
        <meta property="og:type" content="website" />
      </Helmet>
      <div className="flex flex-col items-center justify-center h-screen">
        <h1>{metaData.title}</h1>
      </div>
    </>
  );
};

export default Top;
```

**Comment by icyJoseph** [comment]
I can see the open-graph data in:

- https://web-toolbox.dev/tools/ogp-checker
- https://www.linkedin.com/post-inspector/inspect/https:%2F%2Frakko.tools%2Ftools%2F9%2F 

No problem. Did you fix it?

**Reply by SuzukiMika-dev** [reply]
@icyJoseph 
Thank you very much for checking!
I'm sorry. It was my omission.
Click here for the URL of the site where you want to display OGP.
https://main.dx07kywe7vsmk.amplifyapp.com/top/

I tried searching on the analysis site you provided, but both returned errors.

**Reply by icyJoseph** [reply]
https://main.dx07kywe7vsmk.amplifyapp.com/top/ returns errors when navigating to it... 

So it is not even working to visit 🤔

**Reply by SuzukiMika-dev** [reply]
@icyJoseph 

Dear Mr.icyJoseph, Thank you for your reply. We apologize for the delay in getting back to you.
When I imported Head for NextJS from next/head and used it instead of react-helmet-async, it worked as expected.

Thank you for taking the time out of your busy schedule to check and reply🙏

**Comment by SuzukiMika-dev** [comment]
I found out the reason why OGP is not displayed.
I used Head for NextJS instead of react-helmet-async after the error and it worked.
The Head tag is not imported from next/document, but from next/head.

```
import React from "react";
import Head from "next/head"

export const getServerSideProps = async () => {
  const [res] = await Promise.all([
    fetch(
      `https://gist.githubusercontent.com/SuzukiMika-dev/55c0279af147f2f4ca2da5c2e7c8d084/raw/meta-movie-info.json`
    ),
  ]);
  if (!res.ok) {
    const errorText = await res.text();
    console.error("Error fetching data:", errorText);
    return;
  }
  const data = await res.json();

  return {
    props: {
      metaData: data,
    },
  };
};

const Top: React.FC<{ metaData: any }> = ({ metaData }) => {
  return (
    <>
      <Head>
        <title>{metaData.title}</title>
        <meta name="description" content={metaData.description} />
        <meta property="og:site_name" content={metaData["og:site-name"]} />
        <meta property="og:image" content={metaData["og:image"]} />
        <meta property="og:title" content={metaData["og:title"]} />
        <meta property="og:description" content={metaData["og:description"]} />
        <meta property="og:url" content={metaData["og:url"]} />
        <meta property="og:type" content="website" />
      </Head>
      <div className="flex flex-col items-center justify-center h-screen">
        <h1>{metaData.title}</h1>
      </div>
    </>
  );
};

export default Top;
```
